lib/rubocop/cop/style/it_assignment.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for assignments to a local `it` variable inside a block
+      # where `it` can refer to the first anonymous parameter.
+      #
+      # Although Ruby allows reassigning `it` in these cases, it could
+      # cause confusion if `it` is used as a block parameter elsewhere.
+      #
+      # @example
+      #   # bad
+      #   foo { it = 5 }
+      #
+      #   # good
+      #   foo { bar = 5 }
+      #
+      #   # good - `it` is not a parameter in a block with explicit parameters
```

Comment:
I think it makes sense to ban these as well for consistency, the second paragraph you wrote applies here too

Reply:
Yeah I thought about it, I'm not sure if it makes sense to have conservative/aggressive styles (right now it only affects `it` in blocks that it can be a parameter), or just do it across the board always.

At the very least these examples should likely be "okay" and not "good".

Reply:
I would not add styles initially, can always be added later when there is demand for it. I'd choose aggressive as the only option for now but don't feel too strongly about it.

Reply:
You're right, I've updated the cop to consider numblocks and blocks with parameters as well. Thanks!